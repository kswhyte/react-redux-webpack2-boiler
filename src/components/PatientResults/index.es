import Component from './PatientResults';
export default Component;
