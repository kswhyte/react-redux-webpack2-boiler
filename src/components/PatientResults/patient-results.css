@import "imports/_helpers.css";

.patient-results-wrapper {

    & h3 {
        color: var(--light-grey);
    }

    & .patient-table {
        margin-top: 20px;
        width: 100%;
    }

    & .patient-table th {
        padding: 0px 10px 10px;
        font-family: Open Sans;
        font-size: 12px;
        font-weight: bold;
    }

    & .patient-table td:first-child {
        width: 25%;
    }
    & .patient-table td:not(first-child) {
        width: 12%;
    }
    & .patient-table td:last-child {
        width: 20%;
    }
    & .patient-table td {
        font-family: Open Sans;
        font-size: 12px;
        padding: 13px 10px 13px;
        empty-cells: show;
    }
    & .patient-table tr:not(.no-line) {
        border-bottom: solid 2px var(--light-grey);
    }

    & td > button {
        background: var(--white);
        color: var(--primary-color);
        border: 1px solid var(--primary-color);
        position: relative;
        display: inline-block;
        box-sizing: border-box;
        padding: 6px 10px;
        font-family: 'Open Sans';
        font-size: 10px;
        line-height: 1;
        text-decoration: none;
        text-align: center;
        cursor: pointer;
        border-radius: 0px;
        width: 130px;
        height: 30px;
    }

    & td > button:hover {
        background: var(--primary-color);
        color: var(--white);
        border: 1px solid var(--primary-color);
    }
}