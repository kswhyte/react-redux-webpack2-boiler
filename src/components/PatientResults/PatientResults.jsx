import React from 'react';
import { PropTypes } from 'prop-types';
import Moment from 'moment';

import './patient-results.css';

const propTypes = {
    patientSearchResults: PropTypes.array
};

const PatientResults = (props) => {
  return (
    <div className="patient-results-wrapper sm-spacer">
        <table className="patient-table">
          <thead>
            <tr>
              <th>Full Name</th>
              <th>Date of Birth</th>
              <th>Phone</th>
              <th>Zip</th>
              <th>Modified</th>
              <th>Actions</th>
            </tr>
          </thead>
          <tbody>
            {props.patientSearchResults.map(person => {
              return (
                <tr key={person.patientId} >
                  <td>{person.firstName} {person.lastName}</td>
                  <td>{Moment(person.dateOfBirth).format('MMMM Do, YYYY')}</td>
                  <td>{person.phone}</td>
                  <td>{person.zipCode}</td>
                  <td>{person.modified}</td>
                  <td><button>VIEW THIS RECORD</button></td>
                </tr>
              );
            })}
          </tbody>
        </table>
    </div>
  );
};

PatientResults.propTypes = propTypes;

export default PatientResults;
