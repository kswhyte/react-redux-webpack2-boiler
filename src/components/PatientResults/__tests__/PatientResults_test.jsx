import React from 'react';
import PatientResults from '../PatientResults';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import { mount } from 'enzyme';
import expect from 'expect';

const stories = storiesOf('PatientResults', module);

stories.add('Story', () => {
  const patientResultsStory = (
        <PatientResults />
    );

  specs(() =>
    describe('Show up', () => {
      it('Should render the PatientResults component without crashing', () => {
        let output = mount(patientResultsStory);
        expect(output.find('.patient-results-wrapper').length).toEqual(1);
      });
    })
  );
  return patientResultsStory;
});
