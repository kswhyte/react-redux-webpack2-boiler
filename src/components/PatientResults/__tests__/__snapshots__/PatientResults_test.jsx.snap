// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientResults 1`] = `
<div
  className="patient-results-wrapper sm-spacer"
>
  <table
    className="patient-table"
  >
    <thead>
      <tr>
        <th>
          Full Name
        </th>
        <th>
          Date of Birth
        </th>
        <th>
          Phone
        </th>
        <th>
          Zip
        </th>
        <th>
          Modified
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Julialemon
           
          Sugerhillgang
        </td>
        <td>
          February 27th, 1900
        </td>
        <td>
          720-771-7277
        </td>
        <td>
          80202
        </td>
        <td>
          Jun 16, 2017
        </td>
        <td>
          <button>
            VIEW THIS RECORD
          </button>
        </td>
      </tr>
      <tr>
        <td>
          Julialemon
           
          Sugarhillgang
        </td>
        <td>
          February 27th, 1900
        </td>
        <td>
          720-771-7277
        </td>
        <td>
          80202
        </td>
        <td>
          Jun 16, 2017
        </td>
        <td>
          <button>
            VIEW THIS RECORD
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
